src/services/document-service.js

```diff
@@ -246,6 +246,31 @@ class DocumentService {
     }
   }
 
+  async hardDeleteDocument(documentKey) {
+    const document = await this.getDocumentByKey(documentKey);
+
+    if (!document.origin.startsWith(DOCUMENT_ORIGIN.external)) {
+      throw new Error(`Only external documents can be hard deleted. Document '${documentKey}' has origin '${document.origin}'`);
+    }
+
+    let lock;
+    try {
+      lock = await this.documentLockStore.takeLock(documentKey);
+
+      logger.info(`Hard deleting external document '${documentKey}'`);
+
+      await this.transactionRunner.run(async session => {
+        await this.documentRevisionStore.deleteMany({ key: documentKey }, { session });
```

Comment:
Hmm, what about the external users? I guess they could be used in other docs though.

Reply:
I thought about that but didn't want to take it too far into sanitizing here. If that becomes an issue with external users that are not referenced, it's something we can clean up when we do https://educandu.atlassian.net/browse/EDU-158